source/client/service_main.cc

```diff
@@ -49,13 +67,53 @@ void ServiceMain::start() {
     absl::StrAppend(&listener_bound_address_, listener_port_);
   }
   ENVOY_LOG(info, "Nighthawk grpc service listening on: {}", listener_bound_address_);
+  if (listener_output_path_ != "") {
+    std::ofstream myfile(listener_output_path_);
+    if (myfile.is_open()) {
+      myfile << listener_bound_address_;
+      myfile.close();
+    }
+  }
   channel_ = grpc::CreateChannel(listener_bound_address_, grpc::InsecureChannelCredentials());
   stub_ = std::make_unique<nighthawk::client::NighthawkService::Stub>(channel_);
+  pipe_fds_.resize(2);
```

Comment:
I realize that signal handlers are a horrible execution environment, but is a pipe the best that is practical? We can't lock etc., and we dont' want to write to shared memory. Maybe we should have a dispatcher and just post to it?

Reply:
Some arguments I weighed with respect to the pipe approach:

Pro:
- easy to verify that it is signal safe

Con:
- Need to verify the dispatcher / post implementation to make sure it restricts to the list of safe calls over at http://man7.org/linux/man-pages/man7/signal-safety.7.html
- Generally `foreign` code may change; is the dispatcher post routine guaranteeing signal safety? Will it remain so if a different event system is swapped in? 
- Lastly we don't have a dispatcher around here today (see my earlier [comment](https://github.com/envoyproxy/nighthawk/pull/214#discussion_r356139568))

Reply:
Fair enough, let's go with this, we can clean up later as needed.